src/routes/recovery/guards/disable-recovery-alerts.guard.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { verifyMessage } from 'viem';
+
+/**
+ * The DisableRecoveryAlertsGuard guard should be used on routes that require
+ * authenticated actions for deleting recovery contracts.
```

Comment:
We are not deleting the recovery contracts with this signature. We are disabling the alerts that can be triggered by a recovery contract.

Reply:
Technically we are deleting the contracts from Tenderly. It should match its name though so I adjusted the comment in cab30bf4823a04b278ad156652bfe0b319df3044.

Reply:
Yes it's more like deleting from a registry that is used for Alerts. I think "that require authenticated actions for deleting recovery contracts" can be misleading in the context of smart contract.